src/khoj/configure.py

```diff
@@ -303,9 +304,16 @@ def configure_middleware(app):
     app.add_middleware(SessionMiddleware, secret_key=os.environ.get("KHOJ_DJANGO_SECRET_KEY", "!secret"))
 
 
-@schedule.repeat(schedule.every(22).to(26).hours)
+@schedule.repeat(schedule.every(22).to(25).hours)
 def update_search_index():
     try:
+        if ProcessLockAdapters.is_process_locked(ProcessLock.Operation.UPDATE_EMBEDDINGS):
+            logger.info("🔒 Skipping update search index due to lock")
+            return
+        ProcessLockAdapters.set_process_lock(
+            ProcessLock.Operation.UPDATE_EMBEDDINGS, max_duration_in_seconds=60 * 60 * 12
```

Comment:
Should the max duration of the lock be shorter than 12 hours? I'd have expected something closer in the range of a few minutes (12-30mins)

Reply:
Hmmmmmm, I actually don't know how long the complete indexing job generally takes. But maybe we can set it to 2 or 3 hours to be safe. I'll also add a timer here so we can collect some information about the expected run time of this job.

Reply:
And in this range, I'd rather err on the side of there being a single job that runs a long time, rather than risk having multiple workers spin off the same indexing job again.